[BUG] Crash when space link has an specific expiration date format

### Steps to reproduce

1.  In iOS client, create a link over a space, and add an expiration date
2.  In Android client, open `Spaces`
3. On the space containing the link, open `Members`


### Actual behaviour

app crashes, this the stack trace:

```
               Caused by: java.text.ParseException: Unparseable date: "2026-05-20T08:42:00Z"
                 	at java.text.DateFormat.parse(DateFormat.java:400)
                 	at com.owncloud.android.utils.DisplayUtils.displayDateToHumanReadable(DisplayUtils.java:293)
                 	at com.owncloud.android.presentation.spaces.links.SpaceLinksAdapter.onBindViewHolder(SpaceLinksAdapter.kt:63)
                 	at com.owncloud.android.presentation.spaces.links.SpaceLinksAdapter.onBindViewHolder(SpaceLinksAdapter.kt:36)
                 	at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7678)
                 	at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7761)
                 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6582)
                 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6848)
                 	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6688)
                 	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:6684)
                 	at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2362)
                 	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1662)
                 	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1622)
                 	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:687)
                 	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:4604)
                 	at androidx.recyclerview.widget.RecyclerView.onMeasure(RecyclerView.java:3981)
                 	at android.view.View.measure(View.java:29423)
                 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7068)
                 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1608)
                 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:878)
                 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
                 	at android.view.View.measure(View.java:29423)
                 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7068)
                 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1608)
                 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:878)
                 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
                 	at android.view.View.measure(View.java:29423)
                 	at androidx.core.widget.NestedScrollView.measureChildWithMargins(NestedScrollView.java:1917)
                 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                 	at androidx.core.widget.NestedScrollView.onMeasure(NestedScrollView.java:642)
                 	at android.view.View.measure(View.java:29423)
                 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7068)
                 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                 	at android.view.View.measure(View.java:29423)
                 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7068)
                 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1608)
                 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:878)
                 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:721)
                 	at android.view.View.measure(View.java:29423)
                 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7068)
                 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
                 	at androidx.appcompat.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:145)
                 	at android.view.View.measure(View.java:29423)
                 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:7068)
                 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
```

and this is how the server response looks like:

```            
            "createdDateTime": "2026-04-14T16:17:15.732395048Z",
            "expirationDateTime": "2026-05-20T08:42:00Z",
            "hasPassword": true,
            "id": "ohDmhbttvsSCFyI",
            "link": {
                "@libre.graph.displayName": "Name-test",
                "@libre.graph.quickLink": false,
                "preventsDownload": false,
                "type": "view",
                "webUrl": "https://xx.xx.xx.xx:9200/s/AggnLbATdNudcUU"
            }
        }
```

If the expiration date was created in the web, Android app does not crash. Same response fragment in that case:

```
            "createdDateTime": "2026-04-14T16:17:15.732395048Z",
            "expirationDateTime": "2026-05-20T21:59:59.999Z",
            "hasPassword": true,
            "id": "ohDmhbttvsSCFyI",
            "link": {
                "@libre.graph.displayName": "Name-test",
                "@libre.graph.quickLink": false,
                "preventsDownload": false,
                "type": "view",
                "webUrl": "https://xx.xx.xx.xx:9200/s/AggnLbATdNudcUU"
            }
        }
```

### Expected behaviour

No crash.
 
Xiaomi Redmi 13 Note, Android 15
commit: `6ac162bd1`

--- a/changelog/unreleased/4783
+++ b/changelog/unreleased/4783
@@ -5,3 +5,4 @@ This new section shows information like public link names, types and expiration 
 
 https://github.com/owncloud/android/issues/4752
 https://github.com/owncloud/android/pull/4783
+https://github.com/owncloud/android/pull/4834

--- a/owncloudApp/src/main/java/com/owncloud/android/utils/DisplayUtils.java
+++ b/owncloudApp/src/main/java/com/owncloud/android/utils/DisplayUtils.java
@@ -62,6 +62,7 @@ public class DisplayUtils {
     private static final int[] sizeScales = {0, 0, 1, 1, 1, 2, 2, 2, 2};
 
     private static final String DATE_FORMAT_DISPLAY = "dd/MM/yyyy HH:mm";
+    private static final String DATE_FORMAT_ISO_WITHOUT_MILLISECONDS = "yyyy-MM-dd'T'HH:mm:ss'Z'";
     public static final String DATE_FORMAT_ISO = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
 
     private static Map<String, String> mimeType2HumanReadable;
@@ -290,8 +291,12 @@ public class DisplayUtils {
     public static String displayDateToHumanReadable(String date) throws ParseException {
         SimpleDateFormat parser = new SimpleDateFormat(DATE_FORMAT_ISO, Locale.ROOT);
         parser.setTimeZone(TimeZone.getTimeZone("UTC"));
-        Date dateParsed = parser.parse(date);
         SimpleDateFormat formatter = new SimpleDateFormat(DATE_FORMAT_DISPLAY, Locale.ROOT);
-        return formatter.format(dateParsed);
+        try {
+            return formatter.format(parser.parse(date));
+        } catch (ParseException e) {
+            parser.applyPattern(DATE_FORMAT_ISO_WITHOUT_MILLISECONDS);
+            return formatter.format(parser.parse(date));
+        }
     }
 }

